### PR TITLE
[ng] toh-4 page updates: we --> you, etc

### LINKED ISSUE #640: [ng] toh-4 sync needed to bring in the last tech writer edits
E.g., toh-4 still uses "we" vs "you".

### PR DESCRIPTION
**Pull request by chalin**
Fixes #640
- Sync with tech writer's sweep of toh-4
- Added section on "Use `async`/`await`", contrasting with `Future.then()` based version
- Removed repeated details about "Keep the app compiling and running", replacing is by a simpler single sentence in `toh-2` to `-6`, which refers to the full details in `toh-1`.
- Also updated `toh-*` page description metadata

Staged at: https://angulardart-org-dev.firebaseapp.com/angular/tutorial/toh-pt4

### CODE REVIEW

File: src/angular/tutorial/toh-pt1.jade [hunk 1]
```diff
@@ -36,7 +36,18 @@ include ../_util-fns
       .file styles.css
     .file pubspec.yaml
 
-include keep-app-running
+a#keep-app-running
+:marked
+  ## Keep the app compiling and running
```
Comment (kwalrath) [inline src/angular/tutorial/toh-pt1.jade]: The only problem with this text is that people using WebStorm/IntelliJ don't need to run pub serve explicitly. We should acknowledge that. (Actually, it should be the primary use case.)

File: src/angular/tutorial/toh-pt4.jade [hunk 2]
```diff
@@ -50,350 +48,303 @@ include ../_util-fns
...
+  and await][] section of the Dart language tutorial on [Asynchronous
+  Programming: Futures][].
+    
+  [Asynchronous Programming: Futures]: https://www.dartlang.org/tutorials/language/futures
+  [Async and await]: https://www.dartlang.org/tutorials/language/futures#async-and-await
+
 .l-sub-section
   :marked
-    Check out the "[Take it slow](#slow)" appendix to see what the app might be like with a poor connection.
+    At the end of this page, [Appendix: take it slow](#slow) describes what the app might be like with a poor connection.
```
Comment (kwalrath) [inline src/angular/tutorial/toh-pt4.jade]: take -> Take

File: src/angular/tutorial/keep-app-running.jade [hunk 3]
```diff
@@ -1,12 +1,4 @@
...
-
-  Open a terminal/console window.
-  Start the Dart compiler, watch for changes, and start the server by entering the command:
-
-code-example(language="sh" class="code-shell").
-  pub serve
-
-:marked
-  You can keep building the Tour of Heroes without pausing to recompile.
+  Keep the app compiling and running while you build the Tour of Heroes
```
Comment (kwalrath) [inline src/angular/tutorial/keep-app-running.jade]: We need to adjust this, as well, to reflect the common IDE use case.

File: src/angular/tutorial/toh-pt1.jade (hunk 1 above)
Comment (chalin) [inline src/angular/tutorial/toh-pt1.jade]: Fixed. I've reworked this since we already explain how to "run the app" and "keep it running" in the "Setup for development" page, and toh-pt0.

File: src/angular/tutorial/keep-app-running.jade (hunk 3 above)
Comment (chalin) [inline src/angular/tutorial/keep-app-running.jade]: done

File: src/angular/tutorial/toh-pt4.jade (hunk 2 above)
Comment (chalin) [inline src/angular/tutorial/toh-pt4.jade]: fixed

**Comment by chalin** [comment]
Updates done. Staging server refreshed.

File: src/angular/tutorial/keep-app-running.jade [hunk 4]
```diff
@@ -1,12 +1,6 @@
...
-  Open a terminal/console window.
-  Start the Dart compiler, watch for changes, and start the server by entering the command:
-
-code-example(language="sh" class="code-shell").
-  pub serve
-
-:marked
-  You can keep building the Tour of Heroes without pausing to recompile.
+  If it isn't running already, 
+  [run the app](toh-pt0#run-the-app).
```
Comment (kwalrath) [inline src/angular/tutorial/keep-app-running.jade]: Since toh-pt0#run-the-app just links to /angular/guide/setup#run-the-app, let's change the link to be to /angular/guide/setup#run-the-app.

File: src/angular/tutorial/keep-app-running.jade [hunk 5]
```diff
@@ -1,12 +1,6 @@
...
+  As you make changes,
+  [keep it running](toh-pt0#keep-the-app-running)
```
Comment (kwalrath) [inline src/angular/tutorial/keep-app-running.jade]: Perhaps this should link to /angular/guide/setup#reload-the-app?

File: src/angular/tutorial/toh-pt0.jade [hunk 6]
```diff
@@ -22,15 +22,27 @@ include ../_util-fns
...
   and the [angular-examples/quickstart](https://github.com/angular-examples/quickstart)
   GitHub project.
-  For detailed instructions, see [Setup for Development](/angular/guide/setup).
+  For detailed instructions, see 
+  [Create a starter project](/angular/guide/setup#create-a-starter-project)
+  from the [Setup for Development](/angular/guide/setup) page.
+
+  ## Run the app
+
+  Run the app in a manner suitable for your chosen development environment,
```
Comment (kwalrath) [inline src/angular/tutorial/toh-pt0.jade]: This is a bit wordy. How about:

Run the app using your IDE or the command line, ...

File: src/angular/tutorial/keep-app-running.jade [hunk 7]
```diff
@@ -1,4 +1,6 @@
 :marked
-  Keep the app compiling and running while you build the Tour of Heroes
-  by entering the `pub serve` command in a terminal window
-  [as you did before](toh-pt1.html#keep-app-running "Keep the app running").
+  If it isn't running already,
```
Comment (kwalrath) [inline src/angular/tutorial/keep-app-running.jade]: It's unclear what the "it" refers to. Maybe say "the app" instead.

File: src/angular/tutorial/keep-app-running.jade (hunk 4 above)
Comment (chalin) [inline src/angular/tutorial/keep-app-running.jade]: done

File: src/angular/tutorial/toh-pt0.jade (hunk 6 above)
Comment (kwalrath) [inline src/angular/tutorial/toh-pt0.jade]: I just realized that you don't need the section headers if you aren't linking to here. Maybe delete them, moving any info that'd otherwise be missing to the setup page.

File: src/angular/tutorial/keep-app-running.jade (hunk 5 above)
Comment (chalin) [inline src/angular/tutorial/keep-app-running.jade]: done

File: src/angular/tutorial/toh-pt0.jade (hunk 6 above)
Comment (chalin) [inline src/angular/tutorial/toh-pt0.jade]: Yes, better. Done (but changed "using" to "from").

File: src/angular/tutorial/keep-app-running.jade (hunk 7 above)
Comment (chalin) [inline src/angular/tutorial/keep-app-running.jade]: In the second phrase we say to "run the app". This seemed to me like it gave the necessary clarification. Do you agree?

Comment (chalin) [inline src/angular/tutorial/keep-app-running.jade]: Changed it to: 
> If the app isn't running already, [launch the app]().

Comment (kwalrath) [inline src/angular/tutorial/keep-app-running.jade]: It does, but it's suboptimal (grammatically, and for readability's sake) to have a forward reference. Not a big deal.
